processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -1359,6 +1376,44 @@ public ResolvedPair resolveParameterToType(Type declared, Type parameterized) {
         return new ResolvedPair( this, this );
     }
 
+    /**
+     * Resolves generic types using the declared and parameterized types as input.
```

Comment:
would be nice to mention as example an even farther nested structure.

Reply:
added the following example:
```
     * this: List<Optional<T>>
     * declaredType: JAXBElement<T>
     * parameterizedType: JAXBElement<BigDecimal>
     * result: List<Optional<BigDecimal>>
```
hope this is what you meant with nested.

Also figured I'd add some multi-value examples:
```
     * this: Map<K, V>
     * declaredType: JAXBElement<K>
     * parameterizedType: JAXBElement<BigDecimal>
     * result: Map<BigDecimal, V>
     *
     * this: Map<KEY, VALUE>
     * declaredType: HashMap<VALUE, KEY>
     * parameterizedType: HashMap<BigDecimal, String>
     * result: Map<String, BigDecimal>
```

The javadoc preview didn't look that nice so fixed that as well.